### PR TITLE
wallet-api: Move some modules

### PR DESCRIPTION
**Pull request by j-mueller**
* Move validation code to wallet-api, and move the tutorial there too

### CODE REVIEW

File: wallet-api/src/Ledger/Validation.hs [hunk 1]
```diff
@@ -182,8 +195,81 @@ plcDigest = serialise
 -- | Blockchain height
 --   TODO: Use [[Ledger.Height]] when Integer is supported
 data Height = Height
-  { getHeight :: Int
-  } deriving (Eq, Ord, Show, Generic, FromJSON, ToJSON, ToSchema)
+    { getHeight :: Int
+    } deriving (Eq, Ord, Show, Generic, FromJSON, ToJSON, ToSchema)
+
+
+-- | Check if a transaction was signed by a public key
```
Comment (michaelpj) [inline wallet-api/src/Ledger/Validation.hs]: I mildly preferred these in a separate module, but not a big deal.

File: plutus-use-cases/src/Language/PlutusTx/Validation.hs [hunk 2]
```diff
@@ -1,80 +0,0 @@
-module Language.PlutusTx.Validation (
-    module TH
-    ) where
-
-import           Language.PlutusTx.Validation.TH as TH
-
-{- Note [Script types in pending transactions]
```
Comment (michaelpj) [inline plutus-use-cases/src/Language/PlutusTx/Validation.hs]: These notes seem to have been lost, not sure if you want to put them elsewhere?

Comment (j-mueller) [reply plutus-use-cases/src/Language/PlutusTx/Validation.hs]: done. I moved them to their new homes and updated them.

File: wallet-api/src/Ledger/Validation.hs (hunk 1 above)
Comment (j-mueller) [reply wallet-api/src/Ledger/Validation.hs]: I wanted to avoid adding another `TH.hs` file if possible

Comment (michaelpj) [reply wallet-api/src/Ledger/Validation.hs]: Well, in that they previously lived in `Language.PlutusTx.Validation`. On reflection I think it's fine for not all plutus-tx-y stuff to live under `Language.PlutusTx`.

**Pull request merged**